src/caretogether-pwa/src/App.tsx

```diff
@@ -198,40 +129,7 @@ function App() {
     <div className={classes.root}>
       <CssBaseline />
       <Router>
-        <AppBar position="absolute" className={clsx(classes.appBar, (open && !isMobile) && classes.appBarShift)}>
-          <Toolbar className={classes.toolbar} variant="dense">
-            <IconButton
-              edge="start"
-              color="inherit"
-              aria-label="open drawer"
-              onClick={handleDrawerOpen}
-              className={clsx(classes.menuButton, open && classes.menuButtonHidden)}
-            >
-              <MenuIcon />
-            </IconButton>
-            <Typography component="h1" variant="h6" color="inherit" noWrap className={classes.title}>
-              Dashboard
-            </Typography>
-            <div className={classes.search}>
-              <div className={classes.searchIcon}>
-                <SearchIcon />
-              </div>
-              <InputBase
-                placeholder="Search…"
-                classes={{
-                  root: classes.inputRoot,
-                  input: classes.inputInput,
-                }}
-                inputProps={{ 'aria-label': 'search' }}
-              />
-            </div>
-            <IconButton color="inherit">
-              <Badge badgeContent={4} color="secondary">
-                <NotificationsIcon />
-              </Badge>
-            </IconButton>
-          </Toolbar>
-        </AppBar>
+        <Header open={open} handleDrawerOpen={handleDrawerOpen} />
```

Comment:
The biggest change I would say you need to make is to give each section component (`<Dashboard />`, `<Volunteers />`, `<Referrals />`) its own header. Rather than putting a header here that then has to do a bunch of route matching to figure out where it's being used (which breaks the single-responsibility principle).

Or, a bit more advanced React coding but a cleaner and easier-to-maintain end result: wrap each section component with a `<Header />` that has two slots: one for the header items and one for the main page. Use [this technique](https://reactjs.org/docs/composition-vs-inheritance.html#containment).